### PR TITLE
Select Positions - Redeem and Merge

### LINKED ISSUE #79: [FE] - Redeem Position|Merge Positions - Select Position
- Columns to show - TBD
- Add button in Redeem Position
- Create new modal
- Table (similar to condition list) - only Positions where user has balance
- Search and filter
- Select Position on click
- Submit

### PR DESCRIPTION
**Pull request by lmcorbalan**
closes #79 
- Adds Select Position Modal
- Uses Select Position in Redeem Position
- Uses Select Position in Merge Positions

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr153--conditionaltokensexplorer.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr153--conditionaltokensexplorer.review.gnosisdev.com

File: src/components/form/SelectPositions/index.tsx [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    positionIds,
+    positions,
+    updatePositionIds,
+  } = useMultiPositionsContext()
+
+  const {
+    balances,
+    errors: balancesErrors,
+    loading: balancesLoading,
+    updateBalaces,
```
Comment (mariano-aguero) [inline src/components/form/SelectPositions/index.tsx:70]: Wording

File: src/components/form/SelectPositions/index.tsx [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+  positionIds: string[]
+) => {
+  return (
+    !positionsLoading &&
+    !balancesLoading &&
+    positions.length &&
+    balances.length &&
+    balances.length === positionIds.length &&
+    positions.length === positionIds.length &&
+    JSON.stringify(positions.map(({ id }) => id).sort()) === JSON.stringify([...positionIds].sort())
```
Comment (mariano-aguero) [inline src/components/form/SelectPositions/index.tsx:45]: Mmm in your own words, what is the intention of this ?

File: src/components/form/SelectPositions/index.tsx [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+  React.useEffect(() => {
+    if (positionIds.length > 0) {
+      if (
+        canCanculatePositionToDisplay(
+          positionsLoading,
+          balancesLoading,
+          positions,
+          balances,
+          positionIds
+        )
```
Comment (mariano-aguero) [inline src/components/form/SelectPositions/index.tsx:112]: Unless there is something that needs to be done if the first statement is true, regardless of the second, I'd go with both in the same line as it is clearer that both statements need to be true for something to happen

File: src/components/modals/SelectPositionsModal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+
+  const handleSingleAddClick = useCallback((position: Position) => {
+    setSelectedPositions([position])
+  }, [])
+
+  const handleRemoveClick = useCallback((position: Position) => {
+    setSelectedPositions((current) => {
+      return current.filter((selected) => selected.id !== position.id)
+    })
+    // removePositionId(position.id)
```
Comment (mariano-aguero) [inline src/components/modals/SelectPositionsModal/index.tsx:98]: If is not necessary, please remove it

File: src/components/modals/SelectPositionsModal/index.tsx [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+    return [
+      ...defaultColumns.map((col) => ({ ...col, sortable: false })),
+      ...connectedItems.map((col) => ({ ...col, sortable: false })),
+      deleteCell,
+    ]
+  }, [defaultColumns, connectedItems, deleteCell])
+
+  const handleDone = useCallback(() => {
+    if (selectedPositions.length) {
+      if (onConfirm && typeof onConfirm === 'function') {
```
Comment (mariano-aguero) [inline src/components/modals/SelectPositionsModal/index.tsx:197]: Same comments as the other `if`

File: src/contexts/BatchBalanceContext.tsx [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+import { useBalanceForBatchPosition } from 'hooks/useBalanceForBatchPosition'
+import { useErrors } from 'hooks/useErrors'
+import React, { useCallback, useEffect, useState } from 'react'
+
+export interface BatchBalanceContext {
+  positionIds: string[]
+  loading: boolean
+  errors: BalanceErrors[]
+  balances: BigNumber[]
+  updateBalaces: (positionIds: Array<string>) => void
```
Comment (mariano-aguero) [inline src/contexts/BatchBalanceContext.tsx:14]: Wording

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr153--conditionaltokensexplorer.review.gnosisdev.com

File: src/components/form/SelectPositions/index.tsx (hunk 2 above)
Comment (lmcorbalan) [reply src/components/form/SelectPositions/index.tsx:45]: There is a scenario where you change selected positions for different group with the same number of positions, so this checks that all the values are in sync

File: src/components/form/SelectPositions/index.tsx (hunk 3 above)
Comment (lmcorbalan) [reply src/components/form/SelectPositions/index.tsx:112]: This is ok, `positionsIds.length = 0` means that no positions have been selected and `canCanculatePositionToDisplay` checks that everything is in sync to update displayed list

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr153--conditionaltokensexplorer.review.gnosisdev.com

**Comment by gabitoesmiapodo** [comment]
This looks fine for me, I'll let @mariano-aguero approve his last requested changes..

**Pull request merged**